internal/server/middleware.go

```diff
@@ -170,6 +170,8 @@ func unauthenticatedMiddleware(srv *Server) gin.HandlerFunc {
 				c.Request = c.Request.WithContext(data.WithOrg(c.Request.Context(), org))
 				tx.Statement.Context = c.Request.Context() // TODO: remove with gorm
 			}
+			// if this is the first request after sign-up we must exchange the signup cookie for an auth cookie
+			exchangeSignupCookieForSession(c, srv.options.BaseDomain)
```

Comment:
This is currently only being called in the `unauthenticatedMiddleware`, I think we need it in `authenticatedMiddleware` as well. 

We could add this same call to both places, but another option might be to move this into `requireAccessKey`, which is called from both. If we did that, I guess `requireAccessKey` will need another argument, but `exchangeSignupCookieForSession` might be able to get the access key from an arg instead of from `RequestContext`.